### PR TITLE
CLDR-15405 style→formality, make personName order attribute first & most significant

### PR DESCRIPTION
**Pull request by pedberg-icu**
CLDR-15405

- [ ] This PR completes the ticket.

1. Rename personName attribute "style" to "formality"; rename PersonNameFormatter.Style to PersonNameFormatter.Formality
2. Rearrange personName attributes so "order" moves from last (and least significant) to first (and most significant), yielding: order, length, usage, formality
3. Reorder data, SurveyTool display, function parameters, and code chunks to match new personName attribute ordering and naming, for consistency.
4. Add "vi" and "yue" to surnameFirst locales in root and supplementalData.
5. Since #1950 has been merged into main and that updated main merged into this, this also now changes root & English monogram patterns to use -monogram instead of -initial.

There were 3  test failures with  the original version of this that were worked around; those were fixed by Mark is a commit against this PR. There is one new one in the final version of this (TestPersonNameFormatter.java:98 giving "J* B*S*" instead of "JS"), but I have temporarily changed the expected result so the test passes; I think we can merge that way and then fix.

### CODE REVIEW

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java
Comment (pedberg-icu) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java:86]: Without commenting, got error "key is masked by previous values". Not sure what the expectation is here since the parameters duplicate the previous entry.

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java
Comment (pedberg-icu) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java:432]: Adding the allCaps IRENE did not work, we get the Titlecase one. Not sure why.

File: tools/cldr-code/src/test/java/org/unicode/cldr/test/TestExampleGenerator.java
Comment (pedberg-icu) [inline tools/cldr-code/src/test/java/org/unicode/cldr/test/TestExampleGenerator.java:54]: Did not get "Person, Gwen" in the generated example as expected; ExampleGenerator timing issue?

**Review by richgillam — APPROVED** [review]
LGTM.

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java
Comment (macchiati) [reply tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java:86]: Some progress. The problem is in

public NamePatternData(ImmutableMap<ULocale, Order> localeToOrder,
            ImmutableListMultimap<ParameterMatcher, NamePattern> formatParametersToNamePattern) {

In formatParametersToNamePattern.asMap() there should never be two duplicate keys. I think something has gone wrong in the equals() comparison, and will track down.

Comment (macchiati) [reply tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java:86]: The problem was in
            return orders.equals(that.orders)
                && lengths.equals(that.usages)
                && usages.equals(that.lengths)
                && formalities.equals(that.formalities);

somehow the lengths are being compared to usages.

Comment (macchiati) [reply tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java:86]: Fixing in my local copy and going to next one.

Comment (pedberg-icu) [reply tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java:86]: Oops, that was my fault, thanks for tracking down!!

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java
Comment (macchiati) [reply tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java:432]: Progress on this. The problem is that the ExampleGenerator path has hard-coded path dependencies. So I had to fix those.

File: tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleDependencies.java
Comment (pedberg-icu) [inline tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleDependencies.java:266]: @macchiati Thanks for the fixes, but shouldn't the second "style" here also be changed to "formality"? I can make that fix when I reconcile the conflicts.

File: tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleDependencies.java
Comment (pedberg-icu) [inline tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleDependencies.java:269]: And here too.

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java
Comment (pedberg-icu) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java:98]: @macchiati This is the one new unexpected result after I merged in PR-1950 and changed the monogram patterns to use -monogram instead of -initil. I think we can merge with this as is and then fix later, to not hold up getting the other changes in.

File: tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleDependencies.java
Comment (pedberg-icu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleDependencies.java]: Fixed in 3rd commit

File: tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleDependencies.java
Comment (pedberg-icu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleDependencies.java]: Also fixed in 3rd commit.

**Review by macchiati — APPROVED** [review]

**Pull request merged**
